pkg/runner/local_docker.go

```diff
@@ -474,3 +475,61 @@ func (*LocalDockerRunner) ConfigType() reflect.Type {
 func (*LocalDockerRunner) CompatibleBuilders() []string {
 	return []string{"docker:go"}
 }
+
+// This method deletes the testground containers.
+// It does *not* delete any downloaded images or networks.
+// I'll leave a friendly message for how to do a more complete cleanup.
+func (*LocalDockerRunner) TerminateAll() error {
+	log := logging.S()
+	log.Info("Terminate local:docker requested")
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return err
+	}
+	// Build query for testground infrastructure
+	// It seems like you have to do two separate queries for these filter types. If this isn't the
+	// case, this could be made simpler. Do two queries to the local docker daemon. For one, filter
+	// out the names of specific container infrastructure.
+	// For the other filter out where labels match the testground.purpose, which we apply to all
+	// plan containers managed by testground
+	listOpts := types.ContainerListOptions{}
+	listOpts.Filters = filters.NewArgs()
+	listOpts.Filters.Add("name", "testground-sidecar")
+	listOpts.Filters.Add("name", "testground-redis")
+	listOpts.Filters.Add("name", "testground-goproxy")
```

Comment:
The proxy is owned by the `docker:go` builder. We could make builders implement the `Terminator` interface too if needed, but I wouldn't delete it here.